app/src/main/java/nodomain/freeyourgadget/gadgetbridge/impl/GBDevice.java

```diff
@@ -281,15 +286,20 @@ public void setBatteryLevel(short batteryLevel) {
         }
     }
 
-    /**
-     * Returns a string representation of the battery state.
-     */
-    public String getBatteryState() {
-        return mBatteryState != null ? mBatteryState : GBApplication.getContext().getString(R.string._unknown_);
+    public BatteryState getBatteryState() {
+        return mBatteryState;
+    }
+
+    public void setBatteryState(BatteryState mBatteryState) {
+        this.mBatteryState = mBatteryState;
+    }
+
+    public short getBatteryThresholdPercent() {
+        return mBatteryThresholdPercent;
     }
 
-    public void setBatteryState(String batteryState) {
-        mBatteryState = batteryState;
+    public void setBatteryThresholdPercent(short batteryThresholdPercent) {
```

Comment:
This doesn't really belong here (device specific configuration), but I don't know a better place ATM either. DeviceCoordinator would be a logical choice, but that's not available in the service. Let's keep it here for simplicitly, for now.